src/path_control.py

```diff
@@ -1,45 +1,50 @@
-"""
-Helper function to check if path exists
+"""Helper function to check if path exists
 as directory or file. Another helper function
 for opening a downloaded file.
 """
 
 import os
 import platform
+from pathlib import Path
 
 from dotenv import find_dotenv, load_dotenv, set_key
 
 
 def check_path(path, path_type):
-    """Check if path is a directory (path_type = DEFAULT_DIRECTORY)
-    or bib-file (path_type = DEFAULT_BIB_FILE).
-    Note: the path_type is later on passed to other functions.
-    That's we use these names.
+    """Check if path is a directory (path_type = 'dir')
+    or bib-file (path_type = 'bib).
     """
-    if path_type == "DEFAULT_DIRECTORY":
-        return os.path.isdir(path)
-    if path_type == "DEFAULT_BIB_FILE":
-        return os.path.isfile(path) and os.path.splitext(path)[1] == ".bib"
+    path_obj = Path(path)
+    if path_type == "dir":
+        return path_obj.is_dir()
+    if path_type == "bib":
+        return path_obj.exists() and path_obj.suffix == ".bib"
+    return 1
 
 
 def set_default(path, path_type):
-    """Set default directory (path_type = 'DEFAULT_DIRECTORY')
-    or default bib file (path_type = 'DEFAULT_BIB_FILE').
+    """Set default directory (path_type = 'dir')
+    or default bib file (path_type = 'bib').
     """
     if not check_path(path=path, path_type=path_type):
         print("Not a correct path. Please try again.")
+        return 1
```

Comment:
raise YourExceptionType

Reply:
(Also note there are certain exception types that you can inherit from so that `click` ends the program and returns the specified error code)